orangecontrib/spectroscopy/widgets/owspectra.py

```diff
@@ -1151,15 +1137,19 @@ def peak_apply_auto(self, prominence, minHeight, maxHeight, line_overlap):
                 elif not peak_locations:
                     peak_locations.append(val)
                     # used for first value computation as it isn't in the array
-            self.peak_locations = peak_locations
             for i, val in enumerate(peak_locations):
                 self.peak_apply(position=val)
 
     def delete_selected_labels(self):
-        for i, val in enumerate(self.line_list):
+        for i, val in enumerate(self.peak_positions):
```

Comment:
This function does not work for clearing multiple labels: the first deleted one will be correct, all others will not because of the shift. Possible solution: go through peak_position in the reverse order.